Xamarin.Forms.Platform.iOS/CollectionView/ObservableGroupedSource.cs

```diff
@@ -13,76 +12,50 @@ internal class ObservableGroupedSource : IItemsViewSource
 	{
 		readonly UICollectionView _collectionView;
 		readonly UICollectionViewController _collectionViewController;
-		readonly IList _groupSource;
+		readonly IList _originalItemsSource;
 		bool _disposed;
-		SemaphoreSlim _batchUpdating = new SemaphoreSlim(1, 1);
-		List<ObservableItemsSource> _groups = new List<ObservableItemsSource>();
+		readonly List<IItemsViewSource> _groups = new List<IItemsViewSource>();
 
 		public ObservableGroupedSource(IEnumerable groupSource, UICollectionViewController collectionViewController)
 		{
 			_collectionViewController = collectionViewController;
 			_collectionView = _collectionViewController.CollectionView;
-			_groupSource = groupSource as IList ?? new ListSource(groupSource);
+			
+			_originalItemsSource = groupSource as IList ?? new ListSource(groupSource);
 
-			if (_groupSource is INotifyCollectionChanged incc)
+			if (_originalItemsSource is INotifyCollectionChanged incc)
 			{
 				incc.CollectionChanged += CollectionChanged;
 			}
 
-			ResetGroupTracking();
+			ResetGroups();
 		}
 
-		public object this[NSIndexPath indexPath]
-		{
-			get
-			{
-				return GetGroupItemAt(indexPath.Section, (int)indexPath.Item);
-			}
-		}
-
-		public int GroupCount => _groupSource.Count;
+		public object this[NSIndexPath indexPath] => _groups[indexPath.Section][indexPath];
```

Comment:
since all groups are `IItemsViewSource`, they support NSIndexPath, so we can simply use the index operator here.